docs/ReleaseNotes.txt

```diff
@@ -1,3 +1,113 @@
+=== v1.25.0 === (TBD October 2024)
+
+ISPC release featuring new attribute syntax support, extended template handling
+for short vectors and arrays, added support for new Intel GPUs and CPUs, and
+significant code generation enhancements.  This release is based on patched LLVM
+18.1.8, with the minimum required glibc version for Linux binaries raised to
+2.27.
+
+Language changes:
+
+ - Added support for `__attribute__` syntax in variable and function
+   declarations.  The following attributes are now supported: `noescape`,
+  `address_space(N)`, `external_only`, and `unmangled`.
+ - Extended template support for short vectors and array declarations.  You can
+   now use type and non-type parameters to specify the type and dimensions of
+   these types.
+ - Added error messages for variable declarations with incompatible type
+   specifiers.
+ - Supported typedef with same name as struct tag.
+
+Code generation:
+
+ - Switched to the NewGVN pass for global value numbering for CPU targets. This
+   change improves the performance of the generated code in some cases.
+ - Improved generated code for cases when `foreach` loop iteration domain is
+   equal 3.
+ - Reduced the number of copies in dispatcher logic for cases where a user
+   application consists of several multi-target modules.
+ - Propagate LLVM IR flags (e.g., `nsw`, `nuw`) in ISPC optimization passes.
+   This improves the generated code for `foreach` loop counters.
+ - Improved shuffles generation for some AVX2 and AVX512 targets, resulting in
+   up to 80% faster code.
+ - Produced IEEE 754 compliant instructions (`fminnm`/`vminnm`) on ARM for
+   min/max operations.
+
+New targets:
+
+ - Added `xe2hpg-x16` and `xe2hpg-x32` targets for Xe2 Battlemage Intel(R) GPUs.
+ - Added `xe2lpg-x16` and `xe2lpg-x32` targets for Xe2 Lunar Lake Intel(R) GPUs.
+ - Added CPU definitions for `ArrowLake`, `LunarLake` and `GraniteRapids`:
+   `arrowlake`, `lunarlake` and `graniterapids` respectively.
+
+Deprecated targets:
+
+ - `avx512knl-x16`, `gen9-x8` and `gen9-x16` targets are deprecated and will be
+   removed in future releases.
+ - Adjusted device names for `xelpg-x8` and `xelpg-x16` to `mtl-u` and `mtl-h`.
+
+Standard library:
+
+ - Fixed integer overflow in `exp` for -Inf, NaN and big numbers.
```

Comment:
The wording is a bit confusing, as it's floating point `exp()`, while overflow is integer.